### ISSUE TITLE
Crash logs not showing any Stack trace

### ISSUE THREAD

**Issue by Mbhatishere** [getsentry/sentry-cocoa#2569]
### Platform

iOS

### Installed

CocoaPods

### Version

7.31.5

### Steps to Reproduce

My application was originally built in Objective C but now it has Swift and Swift UI code too. Since I have installed Sentry SDK using cocoapods, whenever I try to manually  throw an exception (fatalError()) I only see OutOFMemory log in dashboard with no stack trace. However whenever I try to force an exception in App Delegate file then crash logs appears fine with stack trace. I am not sure if I am missing something here. I have followed the documentation provided on Sentry website. Have added the initialization code in App Delegate.

https://docs.sentry.io/platforms/apple/guides/ios/

     SentrySDK.start {  options in
                options.dsn = 
               "https://b4735bd75cb94e05b8e5393923a0a257@o4504411572994048.ingest.sentry.io/4504411582365696"
                options.debug = true // Enabled debug when first installing is always helpful
                options.beforeSend = { event in
                return event
            }
            
            // Sampling 100% - In Production you probably want to adjust this
            options.tracesSampleRate = 1.0
            options.sessionTrackingIntervalMillis = 5_000

                // Features turned off by default, but worth checking out
                options.enableAppHangTracking = true
                options.enableFileIOTracking = true
                //options.enableCoreDataTracking = true
                options.enableCaptureFailedRequests = true
            options.profilesSampleRate = 1.0
            options.attachScreenshot = true
            options.attachViewHierarchy = true
            if ProcessInfo.processInfo.arguments.contains("--io.sentry.profiling.enable") {
                options.profilesSampleRate = 1
            }
            let httpStatusCodeRange = HttpStatusCodeRange(min: 400, max: 599)
            options.failedRequestStatusCodes = [ httpStatusCodeRange ]
            }

### Expected Result

Crash logs should provide stack trace.

### Actual Result

Crash logs are coming as Out Of Memory in dashboard with no stacktrace.

![image](https://user-images.githubusercontent.com/121812376/210251306-316db803-a897-4517-b3bf-21d2dad28969.png)

**Comment by philipphofmann** [comment]
When do you call `fatalError()`? Do you maybe call it before you start the SentrySDK? 

> try to force an exception in App Delegate

Where exactly do you do that in the app delegate?

**Comment by Mbhatishere** [comment]
> SentrySDK.start {  options in
>             options.dsn = 
>            "https://b4735bd75cb94e05b8e5393923a0a257@o4504411572994048.ingest.sentry.io/4504411582365696"
>             options.debug = true // Enabled debug when first installing is always helpful
>             options.beforeSend = { event in
>             return event
>         }
>         
>         // Sampling 100% - In Production you probably want to adjust this
>         options.tracesSampleRate = 1.0
>         options.sessionTrackingIntervalMillis = 5_000
> 
>             // Features turned off by default, but worth checking out
>             options.enableAppHangTracking = true
>             options.enableFileIOTracking = true
>             //options.enableCoreDataTracking = true
>             options.enableCaptureFailedRequests = true
>         options.profilesSampleRate = 1.0
>         options.attachScreenshot = true
>         options.attachViewHierarchy = true
>         if ProcessInfo.processInfo.arguments.contains("--io.sentry.profiling.enable") {
>             options.profilesSampleRate = 1
>         }
>         let httpStatusCodeRange = HttpStatusCodeRange(min: 400, max: 599)
>         options.failedRequestStatusCodes = [ httpStatusCodeRange ]
>         }

I called fatalError() in a method in App Delegate . I called it after I called start SentrySDK in didFinishLaunching. 

As I said, to test Sentry logging, I am calling fatalError() in other places in the code but the logging (complete stacktrace) is working fine only when I call it in App Delegate . I need to see complete stack trace whenever I call fatalError() from other places in the code. Hope you understood the question.

**Comment by philipphofmann** [comment]
> but the logging (complete stacktrace) is working fine only when I call it in App Delegate

That seems strange to me. Please ensure that you start the SDK before any other calls. If it still doesn't work, please share a small sample project to reproduce the issue. Otherwise, I don't understand what the problem is.

**Comment by Mbhatishere** [comment]
> > but the logging (complete stacktrace) is working fine only when I call it in App Delegate
> 
> That seems strange to me. Please ensure that you start the SDK before any other calls. If it still doesn't work, please share a small sample project to reproduce the issue. Otherwise, I don't understand what the problem is.

If I create a new project and integrate Sentry in it , it is working fine in that project. 
In my current project, There is Login View controller which is defined as root view controller in App Delegate, once login happens, the root view controller is changed to tab bar controller . Stack trace is showing fine for any crash in Login controller and App delegate but when I put an exception in any of the other view controllers linked to tab bars then Out Of Memory exception is coming up. Have you encountered such kind of issue earlier ? Actually my current project is a mix of Objective C, Swift, Swift UI and combine.

![image](https://user-images.githubusercontent.com/121812376/210510214-f9be83e6-9e02-41df-8324-3b47a37a850d.png)

**Comment by philipphofmann** [comment]
> but when I put an exception in any of the other view controllers linked to tab bars

Does this only happen for fatalError or also other types of crashes? Do you get crash reports when calling `SentrySDK.crash()`?

> Have you encountered such kind of issue earlier ?

No that's the first time I see such an issue.

**Comment by Mbhatishere** [comment]
> > but when I put an exception in any of the other view controllers linked to tab bars
> 
> Does this only happen for fatalError or also other types of crashes? Do you get crash reports when calling `SentrySDK.crash()`?
> 
> > Have you encountered such kind of issue earlier ?
> 
> No that's the first time I see such an issue.

To answer your first question, It is happening for all type of crashes. 

Even I just tried SentrySDK.crash() in one of the tab bar controllers but app got hang and crashed after sometime and dashboard is showing Out Of Memory log . Again, I tried SentrySDK.crash() in Login view controller , it shows correct stack trace in dashboard logs but the tab bar controllers are not showing correct crash logs.

**Comment by philipphofmann** [comment]
What was the last version it still worked properly for you, @lokapodo?

> I guess there is no way to track Out Of Memory issues, but not to track other crashes now, right?

No, if you disable crashes, you would get plenty of false OOMs.

**Comment by philipphofmann** [comment]
Maybe this is related to adding `@try @catch` for NSFileHandle with https://github.com/getsentry/sentry-cocoa/pull/2557 released with 7.31.5. Can you, @lokapodo, check if you also have the problem in `7.31.4`?

**Comment by lokapodo** [comment]
Hey, @philipphofmann 

> What was the last version it still worked properly for you, @lokapodo?

We set up Sentry integration quite recently. So the first version we used was `7.30.0`. And it was working the same way: Sentry was sending OOM issues for all crashes.
Together with the team we were wondering, why did we have so many OOM issues. But then we figured out that OOM event was sending instead of any crash.

> Can you, @lokapodo, check if you also have the problem in 7.31.4?

I have checked `7.31.4` version and this issue is reproduced there too.

Just to be clear, an OOM event is sending instead of a crash when: 
```
      options.enableCrashHandler = false
      options.enableOutOfMemoryTracking = true 
```
or just when:
```
      options.enableCrashHandler = false
```
because `enableOutOfMemoryTracking` default value is `true`.
